R/interop-car.R

```diff
@@ -97,41 +97,335 @@ h_get_contrast <- function(object, effect, type = c("II", "III", "2", "3"), tol
+#'   `test.statistic = "Chisq"`.
+#' @param ... Passed to [car::linearHypothesis()] when `test.statistic =
+#'   "Chisq"`.
 #' @inheritParams h_get_contrast
 #'
-#' @details
-#' `Anova` will return `anova` object with one row per variable and columns
-#' `Num Df`(numerator degrees of freedom), `Denom Df`(denominator degrees of freedom),
-#' `F Statistic` and `Pr(>=F)`.
+#' @details `Anova()` will return `anova` object with one row per variable. If
+#'   `test.statistic = "F"`, columns will be `Num Df`(numerator degrees of
+#'   freedom), `Denom Df` (denominator degrees of freedom), `F Statistic` and
+#'   `Pr(>=F)`.
+#'
+#'   If `test.statistic = "Chisq"`, columns will be `Chisq` (the Chi-squared
+#'   test statistic), `Df` (degrees of freedom), and `Pr(>Chisq)` (p-value).
 #'
 #' @keywords internal
-# Please do not load `car` and then create the documentation. The Rd file will be different.
-Anova.mmrm <- function(mod, type = c("II", "III", "2", "3"), tol = sqrt(.Machine$double.eps), ...) { # nolint
-  assert_double(tol, finite = TRUE, len = 1L)
+#'
+#' @examples
+#'
+#' \dontshow{
+#'
+#' Please do not load `car` and then create the documentation. The Rd file will
+#' be different.}
+Anova.mmrm <- function(mod,
+                       type = c("II", "III", "2", "3"),
+                       tol = sqrt(.Machine$double.eps),
+                       test.statistic = c("F", "Chisq"),
+                       vcov. = vcov(mod, complete = FALSE),
+                       singular.ok = FALSE,
+                       ...) { # nolint
+  assert_class(mod, "mmrm")
+  type <- as.character(type)
   type <- match.arg(type)
-  vars <- colnames(attr(terms(mod$formula_parts$model_formula), "factors"))
-  ret <- lapply(
-    vars,
-    function(x) df_md(mod, h_get_contrast(mod, x, type, tol))
-  )
-  ret_df <- do.call(rbind.data.frame, ret)
-  row.names(ret_df) <- vars
-  colnames(ret_df) <- c("Num Df", "Denom Df", "F Statistic", "Pr(>=F)")
-  class(ret_df) <- c("anova", "data.frame")
-  attr(ret_df, "heading") <- sprintf(
-    "Analysis of Fixed Effect Table (Type %s F tests)",
-    switch(type,
-      "2" = ,
-      "II" = "II",
-      "3" = ,
-      "III" = "III"
+  assert_double(tol, finite = TRUE, len = 1L)
+  test.statistic <- match.arg(test.statistic)
+
+  if (test.statistic == "F") {
+    vars <- colnames(attr(terms(mod$formula_parts$model_formula), "factors"))
+    ret <- lapply(
+      vars,
+      function(x) df_md(mod, h_get_contrast(mod, x, type, tol))
     )
-  )
+    ret_df <- do.call(rbind.data.frame, ret)
+    row.names(ret_df) <- vars
+    colnames(ret_df) <- c("Num Df", "Denom Df", "F Statistic", "Pr(>=F)")
+    attr(ret_df, "heading") <-
+      sprintf(
+        "Analysis of Fixed Effect Table (Type %s %s tests)",
+        switch(type, "2" = , "II" = "II", "3" = , "III" = "III"),
+        test.statistic
+      )
+    class(ret_df) <- c("anova", "data.frame")
+  } else {
+    ret_df <-
+      switch(
```

Comment:
code could be more concise along the lines of 
`test_fun <- switch(type, "2" = h_anova_type2_chisq, "3" = h_anova_type3_chisq)`
and then use `test_fun` because the call is the same